src/modules/hip/kernel/tensor_mean.hpp

```diff
@@ -0,0 +1,695 @@
+                                        src_f24.f8[1].f1[3]);                    // perform small work of reducing G float4s to float using 1024 threads and store in Shared
+    partialBSum_smem[hipThreadIdx_x] = (src_f24.f8[2].f1[0] +
+                                        src_f24.f8[2].f1[1] +
+                                        src_f24.f8[2].f1[2] +
+                                        src_f24.f8[2].f1[3]);                    // perform small work of reducing B float4s to float using 1024 threads and store in Shared
+
+    __syncthreads();                                                             // syncthreads after Shared load
+
+    // Reduction of 1024 floats on 1024 threads per block in x dimension
+    for (int threadMax = 512; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRSum_smem[hipThreadIdx_x] += partialRSum_smem[hipThreadIdx_x + threadMax];
+            partialGSum_smem[hipThreadIdx_x] += partialGSum_smem[hipThreadIdx_x + threadMax];
+            partialBSum_smem[hipThreadIdx_x] += partialBSum_smem[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    // Final store to dst
+    if (hipThreadIdx_x == 0)
+    {
+        int totalElements = roiTensorPtrSrc[hipBlockIdx_z].xywhROI.roiHeight * roiTensorPtrSrc[hipBlockIdx_z].xywhROI.roiWidth;
+        float sum = partialRSum_smem[0] + partialGSum_smem[0] + partialBSum_smem[0];
+        int idx = hipBlockIdx_z * 4;
+        dstPtr[idx] = partialRSum_smem[0] / totalElements;
+        dstPtr[idx + 1] = partialGSum_smem[0] / totalElements;
+        dstPtr[idx + 2] = partialBSum_smem[0] / totalElements;
+        dstPtr[idx + 3] = sum  / (totalElements * 3);
+    }
+}
+
+// -------------------- Set 2 - Kernel Executors --------------------
+// Handle U8 datatype
+RppStatus hip_exec_tensor_mean(Rpp8u *srcPtr,
+                               RpptDescPtr srcDescPtr,
+                               Rpp32f *tensorMeanArr,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = (srcDescPtr->w + 7) >> 3;
+    int globalThreads_y = srcDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
+    int gridDim_x = (int) ceil((float)globalThreads_x/LOCAL_THREADS_X);
+    int gridDim_y = (int) ceil((float)globalThreads_y/LOCAL_THREADS_Y);
+    int gridDim_z = (int) ceil((float)globalThreads_z/LOCAL_THREADS_Z);
+
+    if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z;
+        Rpp32u *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<Rpp32u*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
```

Comment:
Having separate executors and so many separate kernels doesn't feel efficient and I think some consolidation can be done for mean/stdDev/sum. Please try to template them

Reply:
I have reused tensor sum for 1st part of mean computation, I don't think any more templating is possible as we need int to handle Rpp8s and uint for Rpp8u, I have used float internally for other types. 
Lets have a call to discuss further if needed.

Reply:
Some consolidation might be possible in either the exec or the kernel by adding some template variables T and U for Rpp8u and a corresponding vector type you have inside